### PR TITLE
Don't add items to the scene graph until they have a valid parent chain

### PR DESCRIPTION
**Pull request by jherico**
This should fix the issues with the log frequently reporting bad transform data

## Testing

Application behavior should be unchanged.  However, when loading up a domain in the current master build, you will frequently see lots of log messages similar to 
`[hifi.shared] getTransform failed for QUuid("{82ee427c-cfdc-49a8-89a1-177e48369c41}")`.  In this build, these should not appear in the log.

### CODE REVIEW

**Comment by cain-kilgore** [comment]
Testing

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3500/HighFidelity-Beta-PR11421-27350bd7deb3334420aa8ff17ce87544c18aeeef.dmg

**Comment by samcake** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3505/HighFidelity-Beta-PR11421-27350bd7deb3334420aa8ff17ce87544c18aeeef.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3505/HighFidelity-Beta-PR11421-27350bd7deb3334420aa8ff17ce87544c18aeeef.exe

**Comment by cain-kilgore** [comment]
# Test Passed
Test | Result
--- | ---
`getTransform failed for QUuid` In this build, these should not appear in the log. | Pass

Wasn't able to find the getTransform output in the log file.

###### Intel i5 7600K @ 4Ghz, 16GB DDR4 2133Mhz Corsair Vengeance, Asus Strix GTX 1050Ti 3GB, HTC Vive

File: libraries/entities-renderer/src/EntityTreeRenderer.cpp [hunk 1]
```diff
@@ -159,6 +159,78 @@ void EntityTreeRenderer::shutdown() {
...
+            }
+        }
+
+
+        if (!processedIds.empty()) {
+            for (const auto& processedId : processedIds) {
+                _entitiesToAdd.erase(processedId);
+            }
+        }
+    }
```
Comment (AndrewMeadows) [inline libraries/entities-renderer/src/EntityTreeRenderer.cpp]: Why walk through `_entitiesToAdd` twice?  Why not get all this work done in one pass?

Comment (jherico) [reply libraries/entities-renderer/src/EntityTreeRenderer.cpp]: I could do it all in one pass, and initially had it that way, but it was ugly and hard to follow.  when iterating over a container in a case where you _might_ remove something, you must use iterators and follow the pattern shown on line 166.  When combined with the loop logic starting on line 176, the loop as a whole becomes much harder to understand and includes multiple early `continue` statements.  

It's possible that doing it all in one loop would be more efficient, but 90% of the time the `_entitiesToAdd` container is going to be empty anyway (it's only populated when a new entity is added to the scene, until that entity is added to the render tree, usually the very next frame unless it is missing a parent item), so I opted for a less efficient but easier to understand loop.
